gateway-ha/src/test/java/io/trino/gateway/ha/router/TestRoutingGroupSelectorRestAPI.java

```diff
@@ -0,0 +1,255 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.router;
+
+import com.google.common.collect.Multimap;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.JsonBodyGenerator;
+import io.airlift.http.client.JsonResponseHandler;
+import io.airlift.http.client.Request;
+import io.airlift.json.JsonCodec;
+import io.trino.gateway.ha.config.RequestAnalyzerConfig;
+import io.trino.gateway.ha.config.RulesRESTConfiguration;
+import io.trino.gateway.ha.router.schema.RoutingGroupRESTApiBody;
+import io.trino.gateway.ha.router.schema.RoutingGroupRESTApiResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.ws.rs.HttpMethod;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static io.airlift.http.client.JsonResponseHandler.createJsonResponseHandler;
+import static io.airlift.http.client.Request.Builder.preparePost;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.trino.gateway.ha.router.RoutingGroupSelector.ROUTING_GROUP_HEADER;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class TestRoutingGroupSelectorRestAPI
+{
+    RequestAnalyzerConfig requestAnalyzerConfig = new RequestAnalyzerConfig();
+    Field fHttpClient;
+    private static final JsonResponseHandler<RoutingGroupRESTApiResponse> ROUTING_GROUP_REST_API_JSON_RESPONSE_HANDLER =
+            createJsonResponseHandler(jsonCodec(RoutingGroupRESTApiResponse.class));
+
+    @BeforeAll
+    void initialize()
+            throws Exception
+    {
+        requestAnalyzerConfig.setAnalyzeRequest(true);
+        fHttpClient = RESTApiRoutingGroupSelector.class.getDeclaredField("httpClient");
```

Comment:
This would be cleaner imo if you added a package-private setter for httpClient, or a constructor that takes an HttpClient, and annotated it with `@VisibleForTesting` from `com.google.common.annotations`

Reply:
Did not know that there was such annotations like this.
Thx for sharing!